src/dev_utils/peer.rs

```diff
@@ -0,0 +1,190 @@
+    hash::{Hash, Hasher},
+};
+
+pub const NAMES: &[&str] = &[
+    "Alice", "Bob", "Carol", "Dave", "Eric", "Fred", "Gina", "Hank", "Iris", "Judy", "Kent",
+    "Lucy", "Mike", "Nina", "Oran", "Paul", "Quin", "Rose", "Stan", "Tina", "Ulf", "Vera", "Will",
+    "Xaviera", "Yakov", "Zaida", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
+];
+
+lazy_static! {
+    static ref PEERS: Vec<PeerId> = NAMES
+        .iter()
+        .map(|name| PeerId::new_with_keypair(name))
+        .collect();
+}
+
+/// **NOT FOR PRODUCTION USE**: Mock type implementing `PublicId` and `SecretId` traits.  For
+/// non-mocks, these two traits must be implemented by two separate types; a public key and secret
+/// key respectively.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct PeerId {
+    id: String,
+    public_key: PublicKey,
+    secret_key: SecretKey,
+}
+
+impl PeerId {
+    pub fn new(id: &str) -> Self {
+        PEERS
+            .iter()
+            .find(|peer| peer.id == id)
+            .cloned()
+            .unwrap_or_else(|| PeerId::new_with_keypair(id))
+    }
+
+    pub fn named_peer_ids() -> &'static [PeerId] {
+        &PEERS
+    }
+
+    fn new_with_keypair(id: &str) -> Self {
+        let (public_key, secret_key) = gen_keypair();
+        Self {
+            id: id.to_owned(),
+            public_key,
+            secret_key,
+        }
+    }
+
+    pub fn id(&self) -> &str {
+        &self.id
+    }
+
+    pub fn from_index(peer_index: usize) -> Option<Self> {
+        NAMES.get(peer_index).map(|name| PeerId::new(name))
+    }
+
+    pub fn pub_key(&self) -> &Self {
+        &self
+    }
+
+    pub fn sec_key(&self) -> &Self {
+        &self
+    }
+}
+
+impl Debug for PeerId {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        write!(formatter, "{}", self.id)
+    }
+}
+
+impl Hash for PeerId {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.id.hash(state);
+        self.public_key.hash(state);
+    }
+}
+
+impl PartialEq for PeerId {
+    fn eq(&self, other: &PeerId) -> bool {
+        self.id == other.id && self.public_key == other.public_key
+    }
+}
+
+impl Eq for PeerId {}
+
+impl PartialOrd for PeerId {
+    fn partial_cmp(&self, other: &PeerId) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for PeerId {
+    fn cmp(&self, other: &PeerId) -> Ordering {
+        self.id.cmp(&other.id)
+    }
+}
+
+impl PublicId for PeerId {
+    type Signature = Signature;
+
+    fn verify_signature(&self, signature: &Self::Signature, data: &[u8]) -> bool {
+        let mut hasher = DefaultHasher::new();
+        hasher.write(data);
+        hasher.write(&self.public_key.0);
+        let hash = hasher.finish().to_le_bytes();
+
+        signature.0[..hash.len()] == hash
+    }
+}
+
+impl SecretId for PeerId {
+    type PublicId = PeerId;
+
+    fn public_id(&self) -> &Self::PublicId {
+        &self
+    }
+
+    fn sign_detached(&self, data: &[u8]) -> Signature {
```

Comment:
Why not use an actual signature here?

Reply:
This was just follow the `SecretID` trait API to provide a function of creates a detached sig of data (as currently used within parsec/routing). 
Actually, this is not being used for the DKG procedure itself.

I decide to keep it just think it might be better to keep consistent with the current trait iterface that being used. so it could be easier to replace the DKG component.
Actually, this `detached signature` stuff can be removed totally, and the DKG process won't be affected.

Reply:
Perhaps go for removing it. I am not sure about all that secret ID trait stuff just yet. It seems overkill (in routing and parsec) to not just use the lib direct when required.

Reply:
Ok, I will remove the `detached signature` stuff then.